src/ZfcUser/Controller/UserController.php

```diff
@@ -348,7 +354,7 @@ public function changeEmailAction()
     public function getUserService()
     {
         if (!$this->userService) {
-            $this->userService = $this->getServiceLocator()->get('zfcuser_user_service');
+            $this->userService = $this->serviceLocator->get('zfcuser_user_service');
```

Comment:
Same comment as everywhere where `->serviceLocator->get()` is called below.

The point of the deprecation is several-fold:

- zend-servicemanager v3 *no longer defines the `ServiceLocatorAwareInterface`*.
- when zend-mvc v3 drops, *it will no longer provide the `ServiceLocatorAware`* initializer, meaning that this *will not get injected at all*
- we are trying to drive developers *away* from composing service locators in their controllers and other code *unless they are context-sensitive locators* (such as plugin managers). As such, to properly prepare for the removal of the initializer, you should be updating your factories to inject all dependencies, and creating factories to do so if none exist yet.

As such, the approaches are:

- Inject all dependencies in the constructor.
- If some are only used on some execution paths, and you do not want the overhead of creating them, define them as lazy services.
- Alternately, split the controller into multiple controllers, each with a single responsibility, which will allow you to reduce required dependencies for each.